src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -33018,8 +33018,15 @@ static void G()
     }
 }";
             var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
-            // https://github.com/dotnet/roslyn/issues/33011: Deconstruction should infer `string?` for `var y`.
-            comp.VerifyDiagnostics();
+            // https://github.com/dotnet/roslyn/issues/33011: Deconstruction should infer `string?` for `var y`. Shouldn't report nullability
+            // mismatches
+            comp.VerifyDiagnostics(
+                // (8,18): warning CS8619: Nullability of reference types in value of type '(string, string?)' doesn't match target type 'string'.
+                //         foreach ((var x, var y) in F())
```

Comment:
These new diagnostics make no sense to me.

Reply:
As mentioned in the prototype comment, this is a spurious warning. Chris is working on a separate fix for tuple types in foreach statements.